### PR TITLE
BPTL Dashboard: Added Kit Assembly Table to Render Kits & Add Save Kit Functionality

### PR DESCRIPTION
**Pull request by Gbarra9**
This PR addresses the following features:
-> Added table for KIT ASSEMBLY PAGE 
-> Functionality to retrieve all kit resources and render resources to table
-> Correctly increment line item number after retrieving data from database and adding kit data
-> Added Submit button to save kit data and add kit data to database
-> Added Cancel Button to clear inputs from input Row

Checklist:
- [x] Code cleanup
- [x] Check for ES Lint warnings
- [ ] Verify test cases
- [x] Check for GIT conflicts
- [ ] Verified unit tests pass with current changes
- [ ] Any dependencies or modules added 
- [x] Attach PoC
- [x] Notes added

Notes: 
Error handling will be needed for future pull requests to the Kit Assembly Page. More refactors will also be included in future pull requests. The current pull request is a the mvp for the Kit Assembly Page.

PoC:

Kit Assembly Page:
Screen Shot 2021-07-28 at 5 57 53 PM

Kit Assembly Page - Alert on Empty Inputs:
Screen Shot 2021-07-28 at 5 58 16 PM

Kit Assembly Page - On Successful Save:
Screen Shot 2021-07-28 at 5 58 42 PM

### CODE REVIEW

File: src/pages/homeCollection/kitAssembly.js [hunk 1]
```diff
@@ -1,21 +1,391 @@
...
+    for (const key in jsonSaveBody) {
+      if (!jsonSaveBody[key]) {
+        alert("One or more inputs are empty!");
+        return;
+      }
+    }
+
+    // ADD DATA to TABLE
+    addRow(jsonSaveBody);
+    function addRow(jsonSaveBody) {
```
Comment (abhinavjonnada82) [inline src/pages/homeCollection/kitAssembly.js:263]: did you define a function & make a function call within the same function (twice)?
I'm not sure if this is a good practice? Can you put the function definitions outside saveItem()?
If you think this is a good practice, lmk?

File: src/pages/homeCollection/kitAssembly.js [hunk 2]
```diff
@@ -1,21 +1,391 @@
...
+        ${jsonSaveBody.collectionCupId}
+    </td>
+    <td>
+        ${jsonSaveBody.collectionCardId}
+    </td>
+</tr>`
+      );
+    }
+    clearRowInputs();
+    function clearRowInputs() {
```
Comment (abhinavjonnada82) [inline src/pages/homeCollection/kitAssembly.js:290]: Over here as well!

File: src/pages/homeCollection/kitAssembly.js (hunk 1 above)
Comment (Gbarra9) [reply src/pages/homeCollection/kitAssembly.js]: This makes sense. Correct me if I am wrong, having the function declarations addRow and clearRowInputs inside saveItem function would be a waste of resources. The user would be clicking on the submit button repeatedly and the two function declarations would be declared multiple times. But from your suggestion and the optimal solution, by having both addRow and clearRowInputs on the outer scope the function is declared only once within the scope of the module and not multiple times. The addRow and clearRowInputs functions are invoked/called multiple times with each button press that calls the saveItem function.

**Comment by abhinavjonnada82** [comment]
Good job on your first PR & MVP completion ⭐.
LGTM!

**Pull request merged**